src/App.tsx

```diff
@@ -1,24 +1,365 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import cn from 'classnames';
+import {
+  SetStateAction,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { Header } from './Components/Header';
+import { Section } from './Components/Section';
+import { Footer } from './Components/Footer';
+import { Todo } from './types/Todo';
+import * as todoService from './api/todos';
+import { Status } from './types/Status';
 
-const USER_ID = 0;
+const USER_ID = 12083;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [erroMessage, setErrorMessage] = useState('');
+  const [statusTodo, setStatusTodo] = useState('');
+  const [tempTodos, setTempTodos] = useState<Todo[] | null>(null);
+  const [isLoad, setIsLoad] = useState(false);
+
+  const myInputRef = useRef<HTMLInputElement>(null);
+
+  const handleClearCompleted = () => {
+    const completeTodos = todos.filter(todo => todo.completed);
+
+    setIsLoad(true);
+    setTempTodos(completeTodos);
+
+    Promise.all(completeTodos.map(delTodo => {
+      return todoService.deleteTodos(delTodo.id)
+        .then(() => {
+          setTodos(currentTodos => currentTodos.filter(todo => todo.id !== delTodo.id));
+        })
+        .catch((err) => {
+          setErrorMessage('Unable to delete a todo');
+          throw err;
+        });
+    }))
+      .catch((err) => {
+        setErrorMessage('Unable to delete a todo');
+        throw err;
+      })
+      .finally(() => {
+        setTempTodos(null);
+        setIsLoad(false);
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+        myInputRef.current?.focus();
+      });
+  };
+
+  function filterTodos() {
+    switch (statusTodo) {
+      case Status.Active:
+        return todos.filter(todo => !todo.completed);
+      case Status.Completed:
+        return todos.filter(todo => todo.completed);
+      case Status.All:
+        return todos;
+      default: return todos;
+    }
+  }
+
+  useEffect(() => {
+    setErrorMessage('');
+    todoService.getTodos(USER_ID)
+      .then((response: SetStateAction<Todo[]>) => setTodos(response))
+      .catch(() => setErrorMessage('Unable to load todos'))
+      .finally(() => setInterval(() => setErrorMessage(''), 3000));
+  }, []);
+
+  function addTodo({
+    title,
+    userId,
+    completed,
+  }: Todo) {
+    setErrorMessage('');
+
+    const newTamperTodo = {
+      title,
+      userId,
+      completed,
+      id: 0,
+    };
+
+    if (!title.trim()) {
+      setErrorMessage('Title should not be empty');
+      setTimeout(() => {
+        myInputRef.current?.focus();
+      }, 0);
+
+      return Promise.resolve();
+    }
+
+    setTempTodo(newTamperTodo);
+
+    return todoService.addTodos({
```

Comment:
it is `addTodo`, not `Todos`
```suggestion
    return todoService.addTodo({
```